src/socean.ts

```diff
@@ -166,6 +170,144 @@ export class Socean {
     return transactionSequence;
   }
 
+  /**
+   * Signs, sends and confirms the transactions required to deposit a stake account
+   * into the Socean stake pool
+   * @param walletAdapter SOL wallet to deposit SOL from
+   * @param stakeAccount The stake account to deposit.
+   *                     Must be active and delegated to a validator in the stake pool.
+   *                     The entire stake account is deposited. To deposit only a portion of it, prefix this instruction
+   *                     with a stake split instruction and deposit the stake account that's split off instead.
+   * @param referrerPoolTokenAccount PublicKey of a scnSOL token account of the referrer for this deposit
+   * @param confirmOptions transaction confirm options for each transaction
+   * @returns the transaction signatures of the transactions sent and confirmed
+   * @throws RpcError
+   * @throws AccountDoesNotExistError if stake pool or main validator stake account for `stakeAccount`'s validator does not exist
+   * @throws WalletPublicKeyUnavailableError
+   * @throws StakeAccountToDepositInvalidError if stake account to deposit does not meet deposit requirements
```

Comment:
maybe we can add a small helpful hint describing the deposit requirements for the people debugging?

Reply:
see the doc comment for `@param stakeAccount`

Reply:
Fair 👍